crates/core/src/db/datastore/traits.rs

```diff
@@ -459,7 +459,11 @@ pub trait TxDatastore: DataRow + Tx {
 pub trait MutTxDatastore: TxDatastore + MutTx {
     // Tables
     fn create_table_mut_tx(&self, tx: &mut Self::MutTxId, schema: TableDef) -> Result<TableId>;
-    fn row_type_for_table_mut_tx(&self, tx: &Self::MutTxId, table_id: TableId) -> Result<ProductType>;
+    fn row_type_for_table_mut_tx<'tx>(
```

Comment:
I think a comment describing the lifetime is justified, why it's that of the transaction and not self.

Why the change from `Result` to `super::Result`?

Also, not exclusively about this PR, but I think this is some evidence about the lack of utility of this trait. Low level representational details leak through for performance reasons. It's not that plausible to me that a different datastore would want to make the same set of choices.